### DISCUSSION TITLE
Editor improvements

### DISCUSSION THREAD

**Discussion by tutis12** [ppy/osu#30490] [Ideas]
Recently I mapped a song with variable timing on lazer editor. I found that the waveform view on the timeline is a really helpful tool. However, it is about 10ms offsync visually. Usually, I try timing to drums and you can easily see when tick marks aligns with the peaks on the waveform graph. Also, a filter should be applied to the waveform graph, so that the drums are more prominent, maybe it could be marked by some color, or just have a higher peak.

I think adding back lower playback speed options to the timing menu would be useful.

Currently, there is no easy way to move all the timing lines by some offset of time.

Didn't find a shortcut for scaling the selection.

Didn't find a way to adjust the hitsound volumes by section.

No way to add bookmarks?

**Comment by bdach** [comment]
> However, it is about 10ms offsync visually.

Will need more information on this (what platform? what track?) because this should not be happening and just saying it does is not enough to fix.

> Also, a filter should be applied to the waveform graph, so that the drums are more prominent, maybe it could be marked by some color, or just have a higher peak.

This is not how audio processing works. You can't *just do* that.

> I think adding back lower playback speed options to the timing menu would be useful.

Why do you require this?

> Currently, there is no easy way to move all the timing lines by some offset of time.

https://github.com/ppy/osu/issues/23105

> Didn't find a shortcut for scaling the selection.

Ctrl-E

> Didn't find a way to adjust the hitsound volumes by section.

https://github.com/orgs/ppy/projects/18?pane=issue&itemId=85831525

> No way to add bookmarks?

https://github.com/ppy/osu/issues/20883

**Reply by peppy** [reply]
> Also, a filter should be applied to the waveform graph, so that the drums are more prominent, maybe it could be marked by some color, or just have a higher peak.
> This is not how audio processing works. You can't just do that.

Hmm, drum beats should be a lighter colour in the waveform. It's split into three colours based on frequency.

**Reply by peppy** [reply]
> However, it is about 10ms offsync visually.

If you are comparing to the waveform on disk, then yes it is. But this is correct and required for osu! beatmaps due to historical reasons.
